src/03-ticket-calculator.js

```diff
@@ -109,7 +237,81 @@ function calculateTicketPrice(ticketData, ticketInfo) {}
     purchaseTickets(tickets, purchases);
     //> "Ticket type 'discount' cannot be found."
  */
-function purchaseTickets(ticketData, purchases) {}
+function purchaseTickets(ticketData, purchases) {
+ 
+  let totalPrice = 0;
+ 
+  let receiptOne = 'Thank you for visiting the Dinosaur Museum!\n-------------------------------------------\n';
+
+  let receiptExtra = '';
+
+  for(let i = 0; i < purchases.length; i++){
+    
+    let initialTotal = 0;
+
+    let extrasArr = [];
+
+    for(let j = 0; j < purchases[i].extras.length; j++){
+
+      let extraPrice = 0;
+
+      if (ticketData.extras.hasOwnProperty(purchases[i].extras[j])){
+
+        extraPrice += ticketData.extras[purchases[i].extras[j]].priceInCents[purchases[i].entrantType]
+       
+        extrasArr.push(ticketData.extras[purchases[i].extras[j]].description)
+
+      } else {
+
+        return `Extra type '${purchases[i].extras[j]}' cannot be found.`
+    
+      }
+
+      initialTotal += extraPrice
+    
+    }
+
+  if(ticketData.hasOwnProperty(purchases[i].ticketType) && ticketData[purchases[i].ticketType].priceInCents.hasOwnProperty(purchases[i].entrantType)){
+
+      initialTotal += Number(ticketData[purchases[i].ticketType].priceInCents[purchases[i].entrantType])
+  
+   } else if (!ticketData.hasOwnProperty(purchases[i].ticketType)){
+
+      return `Ticket type '${purchases[i].ticketType}' cannot be found.`
+
+   } else if (!ticketData[purchases[i].ticketType].priceInCents.hasOwnProperty(purchases[i].entrantType )){
+
+      return `Entrant type '${purchases[i].entrantType}' cannot be found.`
+
+    }  
```

Comment:
Another approach you could have taken for error handling was to leveraging the `calculateTicketPrice` function. If the return is a string then your know it is an error message and return that, if not you would add the price for it ticket to a total .